src/app/+login-page/shibbolethTargetPage/shibboleth.component.ts

```diff
@@ -0,0 +1,28 @@
+import {Component, OnInit} from '@angular/core';
+import {GetJWTafterShibbLoginAction} from '../../core/auth/auth.actions';
+import {Store} from '@ngrx/store';
+import {CoreState} from '../../core/core.reducers';
+import {Observable, of} from 'rxjs';
+
+@Component({
+  selector: 'ds-shibboleth-page',
+  templateUrl: './shibboleth.component.html',
+  styleUrls: ['./shibboleth.component.scss']
+})
+export class ShibbolethComponent implements OnInit {
```

Comment:
Please could you explain to me the reason for this component? Is this necessary?

Reply:
The target parameter of the shibboleth URL  is the URL the SP redirects the user to after a successful shibboleth authentication. In this implementation the target is ..../shibboleth.
This Angular route (/shibboleth) points to this component which dispatches a “GetJWTafterShibbLoginAction” in its ngOnInit() method. This triggers (via an ngrx-effect) a REST-API POST call to “spring-rest/api/authn/login”. If the Shibboleth Authentication process was successful this call triggers a backend response carrying a JWT token and the user gets logged-in.